### ISSUE TITLE
Promises are used incorrectly

### ISSUE THREAD

**Issue by barbu110** [facebook/fbjs#177]
On a basic 

```
$ flow check
```

I get a ton of errors similar to this one:

```
node_modules/fbemitter/node_modules/fbjs/lib/Deferred.js.flow:25
 25:   _promise: Promise;
                 ^^^^^^^ Promise. Application of polymorphic type needs <list of 1 argument>. (Can use `*` for inferrable ones)
```

As far as I see, `Promise` is defined as a template `Promise<+R>` but it's not used correctly in `fbjs`.

Here is my `.flowconfig`:

```
[ignore]
.*react/node_modules/fbjs.*

[include]
./node_modules/fbjs/flow/include

[libs]
./node_modules/fbjs/flow/lib

[options]
module.system=haste

esproposal.class_static_fields=enable
esproposal.class_instance_fields=enable

suppress_type=$FlowIssue
suppress_type=$FlowFixMe
suppress_type=$FixMe

suppress_comment=\\(.\\|\n\\)*\\$FlowFixMe\\($\\|[^(]\\|(\\(>=0\\.\\(3[0-1]\\|[1-2][0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*relay[a-z,_]*\\)?)\\)
suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(>=0\\.\\(3[0-1]\\|[1-2][0-9]\\|[0-9]\\).[0-9]\\)? *\\(site=[a-z,_]*relay[a-z,_]*\\)?)\\)? #[0-9]+
suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy
```

**Comment by zpao** [comment]
Newer versions of fbjs should have that correct (see https://github.com/facebook/fbjs/blob/94c3af40268b1efd204c5ac6a8c4bfd5952f8c81/packages/fbjs/src/core/Deferred.js#L25). Probably just need to update fbemitter.

FWIW, there are other, unrelated, issues with fbjs, Promises, and Flow due to our use of a non-standard Promise def. You're already pulling that lib in via your flowconfig so it's not a big deal for you, but just in case you see other problems.

**Comment by zpao** [comment]
FWIW, I did the upgrade & published fbemitter@2.0.3.
